sos/plugins/etcd.py

```diff
@@ -31,6 +31,7 @@ class etcd(Plugin, RedHatPlugin):
     def setup(self):
         etcd_url = self.get_etcd_url()
 
+        self.add_forbidden_path('/etc/etcd/ca')
```

Comment:
We still get the correct public certs for the host the report is taken on, they are just located in the directory above.

Reply:
All of the contents of /etc/etcd/ca are important to us though[1]. The only think we should explicitly strip is the *.key files. The certs, crl and fragments folders also should be fine to collect. 

[1] 
[root@master-0 cloud-user]# ll /etc/etcd/ca
total 48
-rw-r--r--. 3 root root  1895 Feb  5 15:59 ca.crt
-rw-r--r--. 1 root root  3272 Feb  5 15:59 ca.key
drwx------. 2 root root    48 Feb  5 16:03 certs
drwx------. 2 root root     6 Feb  5 15:59 crl
drwx------. 2 root root    51 Feb  5 15:59 fragments
-rw-r--r--. 1 root root   252 Feb  5 16:03 index.txt
-rw-r--r--. 1 root root    20 Feb  5 16:03 index.txt.attr
-rw-r--r--. 1 root root    20 Feb  5 15:59 index.txt.attr.old
-rw-r--r--. 1 root root   168 Feb  5 15:59 index.txt.old
-rw-r--r--. 1 root root 12547 Feb  5 15:59 openssl.cnf
-rw-r--r--. 1 root root     3 Feb  5 16:03 serial
-rw-r--r--. 1 root root     3 Feb  5 15:59 serial.old

Reply:
@emmahone that is the exact opposite of what @rjhowe has said (that nothing in `/etc/etcd/ca` is useful for a support analyst). So let's break this down:

- Why do you need `ca.crt`?
- Why do you need the certs, crl, and fragments directory?
- Why do you need `openssl.cnf`?
- Why do you need `index.txt*`?

Reply:
The contents are only needed if you want to know the openssl.cnf used to create the CA. If that is needed it can be requested. There are little to no cases where this data is needed. The ca.crt and the server/peer/client certs for the host are all captured in /etc/etcd/

Reply:
@rjhowe That is true. All I am saying is that gathering the .crt in the same fashion we do in the node/master sections of origin.py would suffice here. 

@TurboTurtle The only file I have ever cared about from that list is the ca.crt (to check whether it is valid). I agree with Ryan that the cases where this is needed are far and few between.

Reply:
Can you not validate the `ca.crt` during the run? I.e. issue a command, and store the result of validation in the report, but not the file itself.